tests/akg.py

```diff
@@ -0,0 +1,60 @@
+import os
+import torch
+from transformers import AutoTokenizer
+from optimum.tpu.distributed_model import DistributedModel
+from loguru import logger
+import sys
+
+# Remove default handler
+logger.remove()
+
+# Add a handler to write to file
+logger.add("distributed_model.log", rotation="100 MB", level="DEBUG")
+
+# Add a handler to write to stderr
+logger.add(sys.stderr, level="INFO")
+
+def sample_greedy(logits):
+    next_logits = logits[:, -1]
+    next_token_id = torch.argmax(next_logits, dim=-1)[:, None].int()
+    return next_token_id
+
+def _test_distributed_model_generation(model_id, max_new_tokens=20):
```

Comment:
for tests, please create one test similar to `tests/test_distributed_model.py` (or modify the existing one). To launch it, you can use pytest: `python -m pytest -sv /path/to/test_mytest.py::test_my_test_function`.